src/session.ts

```diff
@@ -1,29 +1,71 @@
 import { Context } from './context'
 import { Middleware } from './types'
 
-export function session<SessionData extends object>({
-  ttl = Infinity,
-  store = new Map<string, { expires: number; session: SessionData }>(),
-  getSessionKey = (ctx: Context) =>
-    ctx.from && ctx.chat && `${ctx.from.id}:${ctx.chat.id}`,
-} = {}): Middleware.ExtFn<Context, { session?: SessionData }> {
-  const ttlMs = ttl * 1000
+export interface Storage<T> {
+  makeKey: (ctx: Context) => Promise<string | undefined>
```

Comment:
e.q:
user has external user id and real business system id remapping
it need can async